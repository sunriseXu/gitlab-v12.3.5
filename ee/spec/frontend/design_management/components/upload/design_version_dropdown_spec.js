import { createLocalVue, shallowMount } from '@vue/test-utils';
import VueRouter from 'vue-router';
import DesignVersionDropdown from 'ee/design_management/components/upload/design_version_dropdown.vue';
import { GlDropdown, GlDropdownItem } from '@gitlab/ui';
import mockAllVersions from './mock_data/all_versions';

const VERSION_ID = 3;
const localVue = createLocalVue();
localVue.use(VueRouter);
const router = new VueRouter();

describe('Design management design version dropdown component', () => {
  let wrapper;

  function createComponent() {
    wrapper = shallowMount(DesignVersionDropdown, {
      propsData: {
        projectPath: '',
        issueIid: '',
      },
      localVue,
      router,
    });

    wrapper.setData({
      allVersions: mockAllVersions,
    });
  }

  afterEach(() => {
    wrapper.destroy();
  });

  const findVersionLink = index => wrapper.findAll('.js-version-link').at(index);

  it('renders design version dropdown button', () => {
    createComponent();

    expect(wrapper.element).toMatchSnapshot();
  });

  it('renders design version list', () => {
    createComponent();

    expect(wrapper.element).toMatchSnapshot();
  });

  describe('selected version name', () => {
    it('has "latest" on most recent version item', () => {
      createComponent();

      expect(findVersionLink(0).text()).toContain('latest');
    });
  });

  describe('versions list', () => {
    it('pushes version id when a version is clicked', () => {
      createComponent();
      wrapper.vm.$router.push(`/designs?version=${VERSION_ID}`);
      const CurrentVersionNumber = wrapper.vm.getCurrentVersionNumber();

      expect(wrapper.find(GlDropdown).attributes('text')).toBe(
        `Showing Version #${CurrentVersionNumber}`,
      );
    });

    it('should have the same length as apollo query', () => {
      createComponent();

      expect(wrapper.findAll(GlDropdownItem).length).toEqual(wrapper.vm.allVersions.length);
    });
  });
});
