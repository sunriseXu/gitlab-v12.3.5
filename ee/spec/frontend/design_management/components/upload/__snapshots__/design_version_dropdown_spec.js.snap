// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Design management design version dropdown component renders design version dropdown button 1`] = `
<gldropdown-stub
  class="design-version-dropdown"
  issueiid=""
  projectpath=""
  text="Showing Latest Version"
  variant="link"
>
  <gldropdownitem-stub>
    <routerlink-stub
      class="d-flex js-version-link"
      event="click"
      tag="a"
      to="[object Object]"
    >
      <div
        class="flex-grow-1 ml-2"
      >
        <div>
          <strong>
            Version 2
            
            <span>
              (latest)
            </span>
          </strong>
        </div>
      </div>
       
      <i
        class="fa fa-check pull-right"
      />
    </routerlink-stub>
  </gldropdownitem-stub>
  <gldropdownitem-stub>
    <routerlink-stub
      class="d-flex js-version-link"
      event="click"
      tag="a"
      to="[object Object]"
    >
      <div
        class="flex-grow-1 ml-2"
      >
        <div>
          <strong>
            Version 1
            
            <!---->
          </strong>
        </div>
      </div>
       
      <!---->
    </routerlink-stub>
  </gldropdownitem-stub>
</gldropdown-stub>
`;

exports[`Design management design version dropdown component renders design version list 1`] = `
<gldropdown-stub
  class="design-version-dropdown"
  issueiid=""
  projectpath=""
  text="Showing Latest Version"
  variant="link"
>
  <gldropdownitem-stub>
    <routerlink-stub
      class="d-flex js-version-link"
      event="click"
      tag="a"
      to="[object Object]"
    >
      <div
        class="flex-grow-1 ml-2"
      >
        <div>
          <strong>
            Version 2
            
            <span>
              (latest)
            </span>
          </strong>
        </div>
      </div>
       
      <i
        class="fa fa-check pull-right"
      />
    </routerlink-stub>
  </gldropdownitem-stub>
  <gldropdownitem-stub>
    <routerlink-stub
      class="d-flex js-version-link"
      event="click"
      tag="a"
      to="[object Object]"
    >
      <div
        class="flex-grow-1 ml-2"
      >
        <div>
          <strong>
            Version 1
            
            <!---->
          </strong>
        </div>
      </div>
       
      <!---->
    </routerlink-stub>
  </gldropdownitem-stub>
</gldropdown-stub>
`;
